python/ray/data/tests/test_dataset.py

```diff
@@ -2224,6 +2224,28 @@ def test_drop_columns(ray_start_regular_shared, tmp_path):
             ds.drop_columns(["dummy_col", "col1", "col2"])
 
 
+def test_select_columns(ray_start_regular_shared):
+    df = pd.DataFrame({"col1": [1, 2, 3], "col2": [2, 3, 4], "col3": [3, 4, 5]})
+    # Test pandas and base cases
+    ds = ray.data.from_pandas(df)
+    assert ds._dataset_format() == "pandas"
+    assert ds.select_columns(columns=["col1", "col2", "col3"]).take(1) == [
+        {"col1": 1, "col2": 2, "col3": 3}
+    ]
+    assert ds.select_columns(columns=["col1", "col2"]).take(1) == [
+        {"col1": 1, "col2": 2}
+    ]
+    assert ds.select_columns(columns=[]).take(1) == [{}]
+    assert ds.select_columns(columns=["col1", "col2", "col2"]).take(1) == [
+        {"col1": 1, "col2": 2}
+    ]
+
+    # Test arrow
+    ds = ds.select_columns(columns=["col1", "col2"], batch_format="pyarrow")
```

Comment:
The "batch_format" should not be a parameter for select_columns().
To create an Arrow format dataset, one way is just to convert the one above, like ds = ds.map_batches(lamda x: x, batch_format="pyarrow"), then the resulted ds should be in Arrow.